sign/bdn/mask.go

```diff
@@ -0,0 +1,113 @@
+package bdn
+
+import (
+	"fmt"
+
+	"go.dedis.ch/kyber/v4"
+	"go.dedis.ch/kyber/v4/sign"
+)
+
+//nolint:interfacebloat
+type Mask interface {
+	GetBit(i int) (bool, error)
+	SetBit(i int, enable bool) error
+
+	IndexOfNthEnabled(nth int) int
+	NthEnabledAtIndex(idx int) int
+
+	Publics() []kyber.Point
+	Participants() []kyber.Point
+
+	CountEnabled() int
+	CountTotal() int
+
+	Len() int
+	Mask() []byte
+	SetMask(mask []byte) error
+	Merge(mask []byte) error
```

Comment:
Strictly speaking, bitmasks don't need to be implemented bytewise so I can probably remove these methods. I'm currently using them to _clone_ the bitmask (https://github.com/Stebalien/kyber/blob/eb8f760ca2dfd944b76103c6484943c6c1a071b3/sign/bdn/mask.go#L87-L103), but I can probably find another way.